modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java

```diff
@@ -152,15 +167,33 @@ else if (Objects.equals(objectField.getDBType(), "Date")) {
 						"nestedFieldArray.value_date#" +
 							objectField.getName()));
 		}
-		else if (Objects.equals(objectField.getDBType(), "Integer")) {
+		else if (Objects.equals(
+					objectField.getDBType(),
+					ObjectFieldConstants.DB_TYPE_INTEGER)) {
+
 			return Optional.of(
 				new IntegerEntityField(
 					objectField.getName(),
 					locale ->
 						"nestedFieldArray.value_integer#" +
 							objectField.getName()));
 		}
-		else if (Objects.equals(objectField.getDBType(), "Long")) {
+		else if (Objects.equals(
+					objectField.getDBType(),
+					ObjectFieldConstants.DB_TYPE_LONG)) {
+
+			if (Objects.equals(
```

Comment:
I wonder if we can just add this condition to the `String` and `Clob` if clause.

Reply:
You're right. I'll make this change.